src/typeconversions.jl

```diff
@@ -64,14 +64,15 @@ end
 
 # These will use finalizers. Close them eagerly to avoid issues.
 datatype(::T) where {T} = Datatype(hdf5_type_id(T), true)
+datatype(T::Type) = Datatype(hdf5_type_id(T), true)
```

Comment:
I believe we actually want this.

```suggestion
datatype(::Type{T}) where T = Datatype(hdf5_type_id(T), true)
```